Generate certificate for other domains

rgoldfinger-quizlet opened the pull request:
<!-- PREMISE: I would really appreciate if you could follow this scheme as much as possible. -->
<!-- Anyway, I believe in open source software and I really appreciate those who decide to work on a project that appreciates. -->
<!-- If this scheme puts you in trouble, feel free to follow it only partially or not to follow it at all. -->
<!-- Your contribution will be taken to heart and welcomed with open arms in the same way, it will only need a little more work to integrate the changes. -->

# Pull Request Details

## Related Issue
Please, open an issue before a pull request. It will help in understanding better the PR aim.

## Types of changes
<!-- Put an `x` in all the boxes that apply -->
- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] I have updated the documentation or my changes dont require it.
- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes (or the existing one were enough).
- [x] All new and existing tests passed.

First, thank you so much for this project, it's a huge win in terms of making it easy to use https for development. 

We would like to use this on a custom domain that we use for dev. It's a pretty simple change to generate the certificate for additional domains, but I'm not super happy with how I've passed the new option down to `mkcert`. Any suggestions for how to improve this?

Very open to writing any tests or documentation that you request.

daquinoaldo commented on certs.js:
```diff
@@ -114,19 +115,19 @@ async function generate(appDataPath = CERT_PATH) {
 }
 
-async function getCerts() {
+async function getCerts(customDomain) {
```
```suggestion
async function getCerts(customDomain = undefined) {
// if customDomain is undefined mkcert uses localhost
```
See (*undefined) at row 100.

daquinoaldo commented on index.js:
```diff
@@ -12,7 +12,7 @@ const getCerts = require(path.resolve(__dirname, "certs.js")).getCerts
 
 // SSL certificate
-const createServer = () => {
+const createServer = (customDomain = "") => {
```
(\*undefined)
```suggestion
const createServer = (customDomain = undefined) => {
```
OR
```suggestion
const createServer = (customDomain) => {
```
OR
```suggestion
const createServer = (domain = "localhost") => {
```
In this latter case change also line 25 from `customDomain` to `domain`.

daquinoaldo commented:
Hi @rgoldfinger-quizlet,
thank you for your interest!

Can you make me an example of some custom domains and why can be useful to trust them on localhost? It could help me in understanding better your feature request.
For example, it seems you generate the certificate for a domain, then you serve the content on localhost (maybe on a custom port). How do you redirect the custom domain to the content? Do we need a proxy?
I am a bit ignorant on the subject, I have never used different domains from localhost on localhost. However, it was one of the features I already wanted to think about, especially with regard to the testing of existing domains on a local PC via a fake DNS or something similar. I wonder if your proposal can open this way.

Meanwhile, I've submitted some changes proposal. Let me know what do you think!

rgoldfinger-quizlet commented:
Thank you for the detailed review. I think I've addressed most of your concerns and bugs that you found. 

Our use case for other domains is mostly that it's much nicer to use a friendly domain when doing local development. It also makes OAuth possible, as some providers restrict using localhost. I think a number of companies use a setup like this -- I've worked at two so far. Something along the lines of `dev.company.com` or `app.dev`, which redirect to `127.0.0.1`. 

I also wanted to mention that the asset server isn't necessary for our case, so we're using `getCerts` directly and using our existing express server.

Does that help?

daquinoaldo commented:
First of all, thank you for the changes to your PR. I'll publish a new release soon.
Thank you also for your example, of course it helps! How do you manage the redirect? Do you simply redirect the domain on 127.0.0.1 for all with DNS or do you use a fake DNS or something similar on localhost?
